### PR TITLE
Update: Don't ignore comments (no-trailing-spaces)

### PR DESCRIPTION
**Pull request by qistoph**
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))

**What rule do you want to change?**
no-trailing-spaces

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior

**Please provide some example code that this change will affect:**

```js
// Comment with trailing space 
```

**What does the rule currently do for this code?**
By default trailing spaces in comments are ignored.

**What will the rule do after it's changed?**
As specified in the docs (https://eslint.org/docs/rules/no-trailing-spaces) the comments are also checked for trailing whitespace by default.

**What changes did you make? (Give an overview)**
- Update the code to invert the check for absence of the options.ignoreComments. The variable ignoreComments will now be **false** if the option is absent, instead of **true**.
- Add test for this rule, focusssing on default behavior.
- Fix 3 instances of this error in lib/rules/generator-star-spacing.js to fix test errors.

**Is there anything you'd like reviewers to focus on?**
If my assumption is correct that the documentation described the preferred behavior.

### CODE REVIEW

File: lib/rules/no-trailing-spaces.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ module.exports = {
 
         const options = context.options[0] || {},
             skipBlankLines = options.skipBlankLines || false,
-            ignoreComments = typeof options.ignoreComments === "undefined" || options.ignoreComments;
+            ignoreComments = typeof options.ignoreComments === "boolean" && options.ignoreComments;
```
Comment (not-an-aardvark) [inline lib/rules/no-trailing-spaces.js]: I think this line can simply be changed to:

```js
ignoreComments = options.ignoreComments;
```

**Comment by kaicataldo** [comment]
This is a breaking change and seems like a common enough error that it could break a lot of people's linting tasks. I know we reserve the right to add this in semver-minor releases, but I'd at least like to discuss as a team and do it intentionally :)

The alternative to this would be to update the documentation for now and defer this change until our next major release (I also think it should be the default behavior).

**Comment by kaicataldo** [comment]
**TSC Summary:**
This is a breaking change for a case that I feel like could be very common.

**Question:**
Do we want to include this in a semver-minor release? Or should we update the documentation to accurately reflect the current behavior and defer this change for our next major release?

**Comment by not-an-aardvark** [comment]
The documentation clearly states the correct behavior, and the current implementation does not follow the documented behavior. In my opinion, this is a clear case where it's just a bugfix in a rule that results in more errors, which would be a semver-minor change according to our semver policy.

Note that this issue was introduced in when the `ignoreComments` option came out in v4.1.0  (which was in June). So I think the ecosystem impact won't be that bad, since it would only potentially affect code which has been modified in the last few months.

**Comment by not-an-aardvark** [comment]
In [today's meeting](https://gitter.im/eslint/tsc-meetings/archives/2017/10/12), the TSC decided to accept this as a semver-minor bugfix.

**Comment by not-an-aardvark** [comment]
Thanks for contributing!

**Pull request merged**